### PR TITLE
Reorder the command line options in the manual

### PR DESCRIPTION
**Pull request by itchyny**
I have always thought the order of command line options in the manual is very unfriendly. It lists `--seq` and `--stream` at the top, but they are used less often than `--null-input` and `--compact-output`. In #2747, I ordered the list carefully to make it easy  to understand, and in this PR, I reordered the manual to match the output of `--help`. Roughly in the order of input options, output options, advanced input/output options (`--stream`, `--seq`), module options, arg options, and other options.

### CODE REVIEW

File: docs/content/manual/manual.yml [hunk 1]
```diff
@@ -104,74 +104,58 @@ sections:
       You can affect how jq reads and writes its input and output
       using some command-line options:
 
-      * `--version`/`-V`:
```
Comment (nicowilliams) [inline docs/content/manual/manual.yml:107]: You don't think `--version` should be first?

**Comment by nicowilliams** [comment]
LGTM!

File: docs/content/manual/manual.yml (hunk 1 above)
Comment (itchyny) [reply docs/content/manual/manual.yml:107]: I think the option (at least the long option) is obvious and don't think it should be first.

**Comment by itchyny** [comment]
Thanks.

**Pull request merged**